server/permalinks.go

```diff
@@ -58,94 +61,111 @@ func (p *Plugin) getReplacements(msg string) []replacement {
 			index: index,
 			word:  word,
 		}
-		// ignore if the word is inside a link
+		// Ignore if the word is inside a link
 		if isInsideLink(msg, index) {
 			continue
 		}
-		// populate the permalinkInfo with the extracted groups of the regex
-		for j, name := range p.gitlabPermalinkRegex.SubexpNames() {
+		// Populate the permalinkInfo with the extracted groups of the regex
+		for j, name := range gitlabPermalinkRegex.SubexpNames() {
 			if j == 0 {
 				continue
 			}
 			switch name {
 			case "haswww":
-				r.permalinkInfo.haswww = m[j]
+				r.permalinkData.haswww = m[j]
 			case "user":
-				r.permalinkInfo.user = m[j]
+				r.permalinkData.user = m[j]
 			case "repo":
-				r.permalinkInfo.repo = m[j]
+				r.permalinkData.repo = m[j]
 			case "commit":
-				r.permalinkInfo.commit = m[j]
+				r.permalinkData.commit = m[j]
 			case "path":
-				r.permalinkInfo.path = m[j]
+				r.permalinkData.path = m[j]
 			case "line":
-				r.permalinkInfo.line = m[j]
+				r.permalinkData.line = m[j]
 			}
 		}
 		replacements = append(replacements, r)
 	}
 	return replacements
 }
 
+func (p *Plugin) replacementOfMessage(r replacement, glClient *gitlab.Client, index int, wg *sync.WaitGroup, myMap map[int]string) {
+	defer wg.Done()
+	// Quick bailout if the commit hash is not proper.
+	if _, err := hex.DecodeString(r.permalinkData.commit); err != nil {
+		p.API.LogDebug("Bad git commit hash in permalink", "error", err.Error(), "hash", r.permalinkData.commit)
+		return
+	}
+
+	// Get the file contents
+	opts := gitlab.GetFileOptions{
+		Ref: &r.permalinkData.commit,
+	}
+	projectPath := fmt.Sprintf("%s/%s", r.permalinkData.user, r.permalinkData.repo)
+	_, cancel := context.WithTimeout(context.Background(), permalinkReqTimeout)
+	file, _, err := glClient.RepositoryFiles.GetFile(projectPath, r.permalinkData.path, &opts)
+	defer cancel()
+	if err != nil {
+		p.API.LogDebug("Error while fetching file contents", "error", err.Error(), "path", r.permalinkData.path)
+		return
+	}
+	// If this is not a file, ignore.
+	if file == nil {
+		p.API.LogWarn("Permalink is not a file", "file", r.permalinkData.path)
+
```

Comment:
remove this line

Reply:
I think it should be there to warn, if permalink is not a file

Reply:
line 116 only it it an empty line